Automattic/newspack-plugin
# Subscriptions Wizard

### All Submissions:






### Changes proposed in this Pull Request:



Closes #53 .
Supersedes #15. 

This is a rather large PR, however it lays the groundwork for all future wizards, so it should be straightforward after this.

Changes:
- Moved the components demos into a new Components Demo wizard (this is mostly why the changed line count is so large)
- Implement the Subscriptions Wizard
- Some small bug fixed for issues blocking development of this feature (I've pointed these out in comments below)

One wizard has a simple pattern (in my opinion):
- On the backend, each wizard maintains its own endpoints in the associated PHP Wizard class.
- On the frontend, each screen is self-contained and manages its own data. The main wizard class exposes a changeScreen method to the screens to enable changing to different screens when buttons are clicked and stuff.

Not implemented in this initial version:
- Loading UI (spinner, etc.) while waiting for API requests. We should discuss how we want loading to behave in general. There is a Muriel design for a loading spinner.
- Plugin requirements handling. See #55. For now **make sure you have WooCommerce, WooCommerce Subscriptions, and WooCommerce Name Your Price active.**

### How to test the changes in this Pull Request:

1. `npm ci; npm run clean; npm run build:webpack`.
2. Go to Newspack > Components Demo and verify that's still working.
3. Go to Newspack > Subscriptions.
4. Add and save subscriptions.
5. Test editing a subscription.
6. Test the "Allow members to specify donation amount" checkbox.
7. Click on the small, round image to see a subscription on the frontend.
8. Test deleting a subscription.

<img width="567" alt="Screen Shot 2019-05-17 at 10 09 49 AM" src="https://user-images.githubusercontent.com/7317227/57945342-7f6c3d80-788e-11e9-8c4e-4776d7c98147.png">
<img width="576" alt="Screen Shot 2019-05-17 at 10 10 24 AM" src="https://user-images.githubusercontent.com/7317227/57945343-8004d400-788e-11e9-8fea-696abac76552.png">


### Other information:
